lib/std/fmt/parse_hex_float.zig

```diff
@@ -202,18 +202,21 @@ pub fn parseHexFloat(comptime T: type, s: []const u8) !T {
         exponent += 1;
     }
 
-    // There are two cases to handle:
-    // - We've truncated more than 0.5ULP (R=S=1), increase the mantissa.
-    // - We've truncated exactly 0.5ULP (R=1 S=0), increase the mantissa if the
-    //   result is odd (G=1).
-    // The two checks can be neatly folded as follows.
-    mantissa |= @boolToInt(mantissa & 0b100 != 0);
-    mantissa += 1;
-
+    // Whenever the guard bit is one (G=1) and:
+    //   - we've truncated more than 0.5ULP (R=S=1)
+    //   - we've truncated exactly 0.5ULP (R=1 S=0)
+    // Were are going to increase the mantissa (round up)
+    var exactly_half = (mantissa & 0b11) == 0b10;
+    var more_than_half = (mantissa & 0b11) == 0b11;
     mantissa >>= 2;
+    var guardBit = mantissa & 1 == 1;
```

Comment:
This is equivalent to `mantissa & true` which is just going to give you a compile error

Reply:
Get the last bit of mantissa as bool.

Reply:
Would something like this be considered more in zig ethos?
```suggestion
    var guardBit = @bitCast(bool, @truncate(u1, mantissa));
```

Reply:
Ah I think I just got confused by order of operations, my mistake. Can you just put parens on it?

Related: #114

Reply:
Oh and probably the simplest way to do this in zig is

```suggestion
    const guard_bit = @truncate(u1, mantissa) != 0;
```